### PR TITLE
autogen: use guile instead of guile@2

### PR DESCRIPTION
**Pull request by iMichka**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/autogen.rb [hunk 1]
```diff
@@ -53,3 +53,78 @@ def install
...
+-	10.[[012]][[,.]]*)
++    darwin*)
++      case ${MACOSX_DEPLOYMENT_TARGET},$host in
++	10.[[012]],*|,*powerpc*)
+ 	  _lt_dar_allow_undefined='$wl-flat_namespace $wl-undefined ${wl}suppress' ;;
+-	10.*)
++	*)
+ 	  _lt_dar_allow_undefined='$wl-undefined ${wl}dynamic_lookup' ;;
+       esac
+     ;;
```
Comment (SMillerDev) [inline Formula/autogen.rb:130]: I think we should keep the flat namespace fix in a separate patch.

Comment (iMichka) [reply Formula/autogen.rb:130]: I had to move it to the patch block, because we need to patch the libtool.m4 file instead of the configure file.

I am adding the patches here: https://github.com/Homebrew/formula-patches/pull/410

File: Formula/autogen.rb [hunk 2]
```diff
@@ -53,3 +53,78 @@ def install
...
++++ autogen-5.18.16/config/guile.m4
+@@ -61,7 +61,7 @@
+ #
+ AC_DEFUN([GUILE_PKG],
+  [PKG_PROG_PKG_CONFIG
+-  _guile_versions_to_search="m4_default([$1], [2.2 2.0 1.8])"
++  _guile_versions_to_search="m4_default([$1], [3.0 2.2 2.0 1.8])"
+   if test -n "$GUILE_EFFECTIVE_VERSION"; then
+     _guile_tmp=""
+     for v in $_guile_versions_to_search; do
```
[the remark this answers is not in the document]
Comment (iMichka) [reply Formula/autogen.rb:105]: No. But we have already shipped a patch for the 2.2 version check some time ago, so the 3.0 version check disabling is almost the same: https://sourceforge.net/p/autogen/bugs/196/
Upstream told that they wanted to remove these checks completely anyway.

Also, I do not expect that much support from upstream:
https://savannah.gnu.org/support/?110504#comment1

> just note I'm retired, so I don't do this for a living anymore.

As autopen is not used that much in homebrew-core, it will probably need to be removed completely in the next years.

File: Formula/autogen.rb [hunk 3]
```diff
@@ -22,17 +22,17 @@ class Autogen < Formula
     sha256 x86_64_linux:   "459c36573772600aab0085300e551ecbbe224a8b036bc10c15d48db1719a5a52"
   end
 
+  depends_on "autoconf" => :build
+  depends_on "automake" => :build
```
Comment (Bo98) [inline Formula/autogen.rb:26]: You don't appear to be using these. Plus for the guile version check your patch is changing `configure` anyway so you probably don't need to patch the m4 files at all.

Comment (iMichka) [reply Formula/autogen.rb:26]: I'm also modifying `config/guile.m4`, and the build complained it needed `aclocal`.

Comment (iMichka) [reply Formula/autogen.rb:26]: The exact errors I had are:
```
/private/tmp/autogen-20211203-68493-1mj787z/autogen-5.18.16/config/missing: line 81: aclocal-1.16: command not found
'aclocal-1.16' is missing on your system.
sh: autom4te: command not found
```

**Comment by Bo98** [comment]
That's because you are modifying `m4` files when you don't need to. It's why our original flat namespace patch only modified `configure` and not the `libtool.m4` file.

This should be sufficient:

```diff
--- autogen-5.18.16.orig/configure
+++ autogen-5.18.16/configure
@@ -14798,7 +14798,7 @@ $as_echo "no" >&6; }
                PKG_CONFIG=""
        fi
 fi
-  _guile_versions_to_search="2.2 2.0 1.8"
+  _guile_versions_to_search="3.0 2.2 2.0 1.8"
   if test -n "$GUILE_EFFECTIVE_VERSION"; then
     _guile_tmp=""
     for v in $_guile_versions_to_search; do
```

**Review by SMillerDev — APPROVED** [review]
